### PR TITLE
Add LEJEPA examples

### PR DESCRIPTION
**Pull request by gabrielfruet**
## Summary

- Adds the LEJEPA example and generated notebook.
- Updates the LEJEPA loss surface to support the example API.

## Testing

- `make format`
- `make generate-example-notebooks`
- `make all-checks` *(fails locally in `tests/utils/test_dist__gather__losses.py::TestGatherLayer_Losses::test_loss_dcl`; the rest passes)*
- Trained the example for 200 epochs; loss went from about 2.0 to 0.3.

### CODE REVIEW

**Review by simonschoelly — APPROVED** [review]
LGTM

File: examples/pytorch/lejepa.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    global_crop_scale=(0.3, 1),
+    local_crop_scale=(0.05, 0.3),
+    gaussian_blur=(0.5, 0.5, 0.5),
+    n_local_views=6,
+)
+
+
+# We ignore object detection annotations by setting target_transform to return 0.
+def target_transform(t):
+    return 0
```
Comment (liopeer) [inline examples/pytorch/lejepa.py]: I think you can just leave this away completely.

File: examples/pytorch/lejepa.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+model.to(device)
+
+dataset = torchvision.datasets.VOCDetection(
+    "datasets/pascal_voc",
+    download=True,
+    transform=transform,
+    target_transform=target_transform,
```
Comment (liopeer) [inline examples/pytorch/lejepa.py]: By default this is anyway `None`.

**Review by liopeer — APPROVED** [review]
Small remark, otherwise LGTM! Feel free to resolve by yourself.

File: examples/pytorch/lejepa.py (hunk 1 above)
Comment (gabrielfruet) [reply examples/pytorch/lejepa.py]: I was wondering if I should remove or not, because it was None anyway

Comment (liopeer) [reply examples/pytorch/lejepa.py]: I don't know why we have this tbh.

Comment (liopeer) [reply examples/pytorch/lejepa.py]: Same for the notebook.
